realm/src/main/java/io/realm/internal/ImplicitTransaction.java

```diff
@@ -54,8 +54,12 @@ public void promoteToWrite() {
 
     public void commitAndContinueAsRead() {
         assertNotClosed();
-        parent.commitAndContinueAsRead();
-        immutable = true;
+        if (!immutable) {
```

Comment:
I prefer below code and no `else`.

```
if (immutable) {
    throw new IllegalStateException("Cannot commit a non-write transaction.");
}
```

because this check is a precondition of the method.